### DISCUSSION TITLE
http request

### DISCUSSION THREAD

**Discussion by otrsz** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#696] [Q&A]
Hello,
i´m trying to connect my spa with a photovoltaic controller to turn on the spa when i have enough power from the sun.
Is it possible to use a http request to turn it on and off?

**Comment by visualapproach** [comment]
Hi, well.. it's possible but needs new code. I'm curious how your setup is working. Can you tell me more about it?

**Comment by otrsz** [comment]
Yes of course, I have a system called openWB running on a raspberry (https://github.com/snaptec/openWB), which asks my PV inverter if there is any power left and controls my wallboxes accordingly. You can also add other devices there, e.g. a heating element in the hot water tank or switchable sockets. However, this doesn't work with the Whirlpool because of the fuse in the cable, which always trips when the power is cut off. But you could also control devices via http command, that's what I was thinking of.

**Comment by visualapproach** [comment]
I see. I will keep in in mind when I have time to fiddle with the code next time

**Reply by Jussiala** [reply]
It's great that you're taking on this work. I think it could work the same as Shelly.
https://shelly.guide/webhooks-https-requests/

**Comment by visualapproach** [comment]
@Jussiala @otrsz 
I made a minor change to the code which would allow to control the tub with http URL requests. It needs to be in the same non-human readable format as the mqtt/ws messages though. For example, to make it beep (clickable link):
[http://layzspa.local/addcommand/?keynamecanbeanything={CMD:13,VALUE:0,XTIME:1715262840,INTERVAL:0,TXT:%22%20beep%22}](http://layzspa.local/addcommand/?keynamecanbeanything={CMD:13,VALUE:0,XTIME:1715262840,INTERVAL:0,TXT:%22%20beep%22})
(The `%22` is just the browser converting special characters.)
You can see what commands (CMD) are available in the docs. Or if you're lazy like me - open the inspection/console on the layzspa config page and see what string you get when adding commands:
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/22987250/2b5264de-697a-4167-a2e4-d5b5bdf5e9b4)

Let me know if this is useful. I don't want to make a special URL for every command due to memory limitations. Making webhooks requires more coding so this was the easiest way to get it done.

PS it's in the development branch DS
